### PR TITLE
[`PLE0101`] error when `__init__` returns a value

### PR DESCRIPTION
**Pull request by r3m0t**
It's my first PR on any rust project 🥳 

#970

### CODE REVIEW

**Comment by sbrugman** [comment]
(@r3m0t Run 'cargo dev generate-all' to update the schema)

File: crates/ruff/src/rules/pylint/rules/return_in_init.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+use super::yield_in_init::in_dunder_init;
+
+define_violation!(
+    pub struct ReturnInInit;
```
Comment (sbrugman) [inline crates/ruff/src/rules/pylint/rules/return_in_init.rs:13]: Please document the rule (you can take [pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/return-in-init.html) as a start)

See https://github.com/charliermarsh/ruff/issues/2646

**Comment by r3m0t** [comment]
Fixed, I had to change `README.md` to check out with LF.

**Comment by sbrugman** [comment]
Couple of minor things.

Also note that the PR title is used for the changelog. The format for rules is `[`code`] description` (we should have PR templates to support this and automatically add the label)

**Comment by r3m0t** [comment]
All addressed I think

**Comment by sbrugman** [comment]
> All addressed I think

Don't see the changes, did you push?

**Comment by r3m0t** [comment]
> > All addressed I think
> 
> Don't see the changes, did you push?

Did you mean the PR title or the commit message itself? Yes, I just force pushed to update the commit message, and added the docs

**Comment by sbrugman** [comment]
Ah sorry let me clarify. There are two more comments: 
https://github.com/charliermarsh/ruff/pull/3007#discussion_r1111047372

**Comment by charliermarsh** [comment]
Welcome! Thanks for the contribution, and thanks to @sbrugman for the review.

**Pull request merged**